staging/src/k8s.io/code-generator/cmd/informer-gen/generators/informer.go

```diff
@@ -123,20 +124,23 @@ type $.type|public$Informer interface {
 var typeInformerStruct = `
 type $.type|private$Informer struct {
 	factory $.interfacesSharedInformerFactory|raw$
+	filter  $.interfacesFilterFunc|raw$
 }
 `
 
 var typeInformerPublicConstructor = `
 // New$.type|public$Informer constructs a new informer for $.type|public$ type.
 // Always prefer using an informer factory to get a shared informer instead of getting an independent
 // one. This reduces memory footprint and number of connections to the server.
-func New$.type|public$Informer(client $.clientSetInterface|raw$$if .namespaced$, namespace string$end$, resyncPeriod $.timeDuration|raw$, indexers $.cacheIndexers|raw$) $.cacheSharedIndexInformer|raw$ {
+func New$.type|public$Informer(client $.clientSetInterface|raw$, filter $.interfacesFilterFunc|raw$, resyncPeriod $.timeDuration|raw$, indexers $.cacheIndexers|raw$) $.cacheSharedIndexInformer|raw$ {
```

Comment:
This change to the function signature will break anyone calling `New*Informer` when they upgrade their copy of client-go. Would it be possible to find another way to do this that isn't a breaking change? Maybe add another constructor?